contacts/api.py

```diff
@@ -0,0 +1,179 @@
+from tokenize import Token
+
+from django.http import HttpResponse
+from django.utils.decorators import method_decorator
+from django.views.decorators.csrf import csrf_exempt
+from rest_framework import permissions, status
+from rest_framework.authtoken.models import Token
+from rest_framework.decorators import api_view, permission_classes
+from rest_framework.permissions import AllowAny
+from rest_framework.status import (
+    HTTP_400_BAD_REQUEST,
+    HTTP_404_NOT_FOUND,
+    HTTP_200_OK
+)
+from rest_framework.response import Response
+from rest_framework.views import APIView
+import datetime
+from contacts.serializers import UserSerializer, ContactSerializer, UserDetailSerializer
+from contacts.models import User, Contact, Email
+from django.contrib.auth import authenticate
+
+
+class UserProfile(APIView):
+    permission_classes = (permissions.AllowAny,)
+
+    def get(self, request):
+        try:
+            if 'token' in request.COOKIES:
+                token = Token.objects.get(key=request.COOKIES['token'])
+            else:
+                token = None
+        except Token.DoesNotExist:
+            token = None
+        if token is not None:
+            pk = request.COOKIES.get('user_id')
+            user = User.objects.get(id=pk)
+            # contacts = self.get_serialized(pk, user)
+            serializer = UserSerializer(user)
+            return Response(serializer.data, status=HTTP_200_OK)
+        else:
+            return Response("Your token expire Login again", status=HTTP_400_BAD_REQUEST)
+
+    def put(self, request):
+        try:
```

Comment:
This token checking code is being duplicated on many API views. please refactor this to make it cleaner.

Reply:
```python
def validate_token(request):
    try:
        if 'token' in request.COOKIES:
            token = Token.objects.get(key=request.COOKIES['token'])
            return token
        else:
            return None
    except Token.DoesNotExist:
        return None

```

Reply:
done